File: cronplusui/src/types/taskLogging.ts
import { TriggerType } from './taskConfig';

export interface TaskLogging {
  id?: string;
  taskType: string;
  triggerType: TriggerType;
  directory: string;
  filePath?: string;
  printerName?: string;
  archiveDirectory?: string;
  sourceFolder: string;
  destinationFolder: string;
  triggeredAt: Date;
  result: string;
  duration?: number; // in milliseconds
  createdAt?: Date;
  updatedAt?: Date;
}

File: cronplusservice/Models/TaskLogging.cs
using System;
using System.Text.Json.Serialization;
using CronPlus.Models;
using SurrealDb.Net.Models;

namespace CronPlus.Models;

public class TaskLogging : Record
{
    public string TaskType { get; set; } = string.Empty;
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TriggerType TriggerType { get; set; }
    public string Directory { get; set; } = string.Empty;
    public string? FilePath { get; set; }
    public string? PrinterName { get; set; }
    public string? ArchiveDirectory { get; set; }
    public string SourceFolder { get; set; } = string.Empty;
    public string DestinationFolder { get; set; } = string.Empty;
    public DateTime TriggeredAt { get; set; } = DateTime.UtcNow;
    public string Result { get; set; } = string.Empty;
    public TimeSpan Duration { get; set; }
}
